### PR TITLE
Changes DatastoreScheduledInterview update() method to use datastore put() and tests its functionality

### PR DESCRIPTION
**Pull request by GrantComm**
Modified the DataStoreScheduledInterviewDao.java file to include a method that changes an entity's fields and added tests for each method in the file. Formatting change in DatastoreAvailabilityDao.java is in pr #100

### CODE REVIEW

**Review by meagancheese — APPROVED** [review]
LGTM

File: src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java [hunk 1]
```diff
@@ -116,11 +115,24 @@ public ScheduledInterview entityToScheduledInterview(Entity scheduledInterviewEn
...
+  /** Updates a scheduledInterview Entity from a scheduledInterview object. */
+  public Entity scheduledInterviewToUpdateEntity(ScheduledInterview scheduledInterview) {
+    Entity scheduledInterviewEntity = new Entity("ScheduledInterview", scheduledInterview.id());
+    setEntityProperties(scheduledInterviewEntity, scheduledInterview);
+    return scheduledInterviewEntity;
+  }
+
+  /** Sets the properties of a scheduledInterview Entity. */
+  static void setEntityProperties(
+      Entity scheduledInterviewEntity, ScheduledInterview scheduledInterview) {
```
Comment (apluscs) [inline src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java]: This function goes against something Alex warned me against. That is, a function that modifies a param instead of returning it. Here it might be fine because it's more explicit in the function name but not sure. We can come back after we get some more info.

**Review by apluscs — APPROVED** [review]
One nit, but LGTM.

File: src/test/java/com/google/sps/data/DatastoreScheduledInterviewTest.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            "user3@mail.com");
+    tester.update(updatedStoredScheduledInterview);
+    Entity updatedEntity = datastore.prepare(new Query("ScheduledInterview")).asSingleEntity();
+    ScheduledInterview updatedScheduledInterview = tester.entityToScheduledInterview(updatedEntity);
+    Assert.assertEquals(updatedStoredScheduledInterview, updatedScheduledInterview);
+  }
+
+  // Tests retrieving a scheduledInterview from Datastore.
+  @Test
+  public void getsScheduledInterview() {
```
Comment (meagancheese) [inline src/test/java/com/google/sps/data/DatastoreScheduledInterviewTest.java:163]: You should add a test where it is not within datastore to make sure an empty Optional is returned.

**Review by meagancheese — CHANGES_REQUESTED** [review]
One more test, otherwise looks good.

File: src/test/java/com/google/sps/data/DatastoreScheduledInterviewTest.java (hunk 2 above)
Comment (GrantComm) [reply src/test/java/com/google/sps/data/DatastoreScheduledInterviewTest.java:163]: Done

File: src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java (hunk 1 above)
Comment (meagancheese) [reply src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java]: So the conclusion is to put this code back within the two methods because creation of the Entity, property setting, and returning should all be within the same method.

**Review by meagancheese — APPROVED** [review]
Small fix (delete setProperties), otherwise LGTM.

File: src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java (hunk 1 above)
Comment (GrantComm) [reply src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java]: Done

File: src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java [hunk 3]
```diff
@@ -123,4 +122,15 @@ public Entity scheduledInterviewToEntity(ScheduledInterview scheduledInterview)
     scheduledInterviewEntity.setProperty("interviewee", scheduledInterview.intervieweeEmail());
     return scheduledInterviewEntity;
   }
+
+  /** Updates a scheduledInterview Entity from a scheduledInterview object. */
+  public Entity scheduledInterviewToUpdateEntity(ScheduledInterview scheduledInterview) {
```
Comment (jalexanderqed) [inline src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java]: I don't think the description here is accurate since this does not actually update anything, just creates a new entity.

Related to that, I think `scheduledInterviewToEntity` or `scheduledInterviewToDatastoreEntity` would be a more appropriate name.

File: src/test/java/com/google/sps/data/DatastoreScheduledInterviewTest.java [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+import java.util.List;
+import java.util.Optional;
+import org.junit.Assert;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class DatastoreScheduledInterviewTest {
+
+  DatastoreScheduledInterviewDao tester = new DatastoreScheduledInterviewDao();
```
Comment (jalexanderqed) [inline src/test/java/com/google/sps/data/DatastoreScheduledInterviewTest.java]: Please give this a more descriptive name than `tester`. Even just `dao` would be appropriate.

**Review by jalexanderqed — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java (hunk 3 above)
Comment (GrantComm) [reply src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java]: It's the method used in to update a stored scheduledInterview entity without losing to generated id. scheduledInterviewToEntity is already a method that creates a scheduledInterview entity with a generated id

File: src/test/java/com/google/sps/data/DatastoreScheduledInterviewTest.java (hunk 4 above)
Comment (GrantComm) [reply src/test/java/com/google/sps/data/DatastoreScheduledInterviewTest.java]: Done

File: src/test/java/com/google/sps/data/DatastoreScheduledInterviewTest.java [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+    Optional<ScheduledInterview> expectedScheduledInterviewOptional =
+        Optional.of(expectedScheduledInterview);
+    Assert.assertEquals(expectedScheduledInterviewOptional, actualScheduledInterviewOptional);
+  }
+
+  // Tests retrieving a scheduledInterview that doesn't exist from Datastore.
+  @Test
+  public void failsGetScheduledInterview() {
+    Optional<ScheduledInterview> actual = dao.get(2);
+    Optional<ScheduledInterview> expected = Optional.empty();
```
Comment (jalexanderqed) [inline src/test/java/com/google/sps/data/DatastoreScheduledInterviewTest.java]: This line is unnecessary. Just do `Assert.assertEquals(Optional.empty(), actual);`

File: src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java (hunk 3 above)
Comment (jalexanderqed) [reply src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java]: In that case, please rename to `scheduledInterviewToEntityForUpdate` or `scheduledInterviewToEntityWithId`. `scheduledInterviewToUpdateEntity` sounds too much like an "UpdateEntity" is a special kind of entity, which is not really accurate.

**Review by jalexanderqed — APPROVED** [review]
